packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -1509,162 +1509,196 @@ class BaseModelSqlv2 {
+                });
+                const cCol = await Column.get({
+                  colId: colOptions.fk_child_column_id,
+                });
+                const readLoader = new DataLoader(async (_ids: string[]) => {
+                  // handle binary(16) foreign keys
+                  const ids = _ids.map((id) => {
+                    if (pCol.ct !== 'binary(16)') return id;
+
+                    //Cast the id to string.
+                    const idAsString = id + '';
+                    // Check if the id is a UUID and the column is binary(16)
+                    const isUUIDBinary16 =
+                      idAsString.length === 36 || idAsString.length === 32;
+                    // If the id is a UUID and the column is binary(16), convert the id to a Buffer. Otherwise, return null to indicate that the id is not a UUID.
+                    const idAsUUID = isUUIDBinary16
+                      ? idAsString.length === 32
+                        ? idAsString.replace(
+                            /(.{8})(.{4})(.{4})(.{4})(.{12})/,
+                            '$1-$2-$3-$4-$5',
+                          )
+                        : idAsString
+                      : null;
+
+                    return idAsUUID
+                      ? Buffer.from(idAsUUID.replace(/-/g, ''), 'hex')
+                      : id;
+                  });
+
+                  const data = await (
+                    await Model.getBaseModelSQL({
+                      id: pCol.fk_model_id,
+                      dbDriver: this.dbDriver,
+                    })
+                  ).list(
+                    {
+                      fieldsSet: (readLoader as any).args?.fieldsSet,
+                      filterArr: [
+                        new Filter({
+                          id: null,
+                          fk_column_id: pCol.id,
+                          fk_model_id: pCol.fk_model_id,
+                          value: ids as any[],
+                          comparison_op: 'in',
+                        }),
+                      ],
+                    },
+                    true,
+                  );
 
-                (readLoader as any).args = args;
+                  const gs = groupBy(data, pCol.title);
+                  return _ids.map(async (id: string) => gs?.[id]?.[0]);
+                });
 
-                return await readLoader.load(this?.[cCol?.title]);
-              };
-              // todo : handle mm
+                // defining HasMany count method within GQL Type class
+                proto[column.title] = async function (args?: any) {
+                  if (
+                    this?.[cCol?.title] === null ||
+                    this?.[cCol?.title] === undefined
+                  )
+                    return null;
+
+                  (readLoader as any).args = args;
+
+                  return await readLoader.load(this?.[cCol?.title]);
+                };
```

Comment:
do we still need this logic in latest version?

Reply:
It's no longer required and I will remove this.